### PR TITLE
Refactor `check_code_format.yml` workflow into an action

### PR DESCRIPTION
**Pull request by ArthurBandaryk**
The link to the issue below describes the reason we do this.
[issue](https://github.com/3rdparty/dev-tools/issues/11)

### CODE REVIEW

**Review by while-false — APPROVED** [review]
LGTM! I thought there was more of a difference, i.e. that an Action was something more. I learned something. Thanks Arthur.

File: action.yml [hunk 1]
```diff
@@ -1,21 +1,31 @@
...
+runs:
+  using: "composite"
+  steps:
     - uses: actions/checkout@v2
+
     - name: Run bootstrap
       run: |
         git submodule init
         git submodule update
         cd dev-tools/ && ./bootstrap
```
Comment (benh) [inline action.yml:15]: Let's leave this up to the caller too!

```suggestion
      run: cd dev-tools/ && ./bootstrap
```

**Review by benh — APPROVED** [review]

File: action.yml [hunk 2]
```diff
@@ -1,21 +1,31 @@
...
-    name: Formatting Check
-    runs-on: ubuntu-latest
-    steps:
+name: 'Check Code Style'
+desciption: 'Check files for correct code style'
+
+runs:
+  using: "composite"
+  steps:
     - uses: actions/checkout@v2
```
[the remark this answers is not in the document]
Comment (ArthurBandaryk) [reply action.yml:9]: As I remember from yesterday talk, we wanna to pass `- uses: actions/checkout@v2` to caller too?

[the remark this answers is not in the document]
Comment (ArthurBandaryk) [reply action.yml:9]: > 

@benh , as I got from the doc we cannot leave this up to the caller, am I right or I'm missing something?

[the remark this answers is not in the document]
Comment (benh) [reply action.yml]: The way I interpreted writing your own action is that you didn't need to check yourself out. Is that correct?

Thus, I'm interpreting `actions/checkout@v2` as trying to checkout the entire enclosing repository (for example, `eventuals`), which we want to leave up to the caller.

Have you tried what you currently have and does it work correctly?

[the remark this answers is not in the document]
Comment (ArthurBandaryk) [reply action.yml]: @benh , yep, I've left `actions/checkout@v2` to the caller and it  worked fine for me. You can check it [here](https://github.com/ArthurBandaryk/composite-action/blob/main/.github/workflows/main.yml)

[the remark this answers is not in the document]
Comment (ArthurBandaryk) [reply action.yml]: Also attach the [link](https://github.com/ArthurBandaryk/dev-tools/blob/main/action.yml) to the composed action in my own repo and the [link](https://github.com/ArthurBandaryk/composite-action/runs/4456705742?check_suite_focus=true) to make you be sure that it works fine.

**Review by while-false — APPROVED** [review]
LGTM. Thanks Arthur. :)

**Pull request merged**